resources/main.mjs

```diff
@@ -0,0 +1,274 @@
+        if (!isNaN(delta)) {
+            const percentDelta = delta * 100 / arithmeticMean;
+            meanSigFig = sigFigFromPercentDelta(percentDelta);
+            formattedDelta = toSigFigPrecision(delta, 2);
+            formattedPercentDelta = toSigFigPrecision(percentDelta, 2) + '%';
+        }
+
+        const formattedMean = toSigFigPrecision(arithmeticMean, Math.max(meanSigFig, 3));
+
+        return {
+            formattedValues: values.map(value =>  {
+                return toSigFigPrecision(value, 4) + ' ' + displayUnit;
+            }),
+            mean: arithmeticMean,
+            formattedMean: formattedMean,
+            formattedDelta: formattedDelta,
+            formattedMeanAndDelta: formattedMean + (formattedDelta ? ' \xb1 ' + formattedDelta + ' (' + formattedPercentDelta + ')' : ''),
+        };
+    }
+
+    _addDetailedResultsRow(table, iterationNumber, value) {
+        const row = document.createElement('tr');
+        const th = document.createElement('th');
+        th.textContent = 'Iteration ' + (iterationNumber + 1);
+        const td = document.createElement('td');
+        td.textContent = value;
+        row.appendChild(th);
+        row.appendChild(td);
+        table.appendChild(row);
+    }
+
+    _updateGaugeNeedle(rpm) {
+        const needleAngle = Math.max(0, Math.min(rpm, 140)) - 70;
+        const needleRotationValue = 'rotate(' + needleAngle + 'deg)';
+
+        const gaugeNeedleElement = document.querySelector('#summarized-results > .gauge .needle');
+        gaugeNeedleElement.style.setProperty('-webkit-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('-moz-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('-ms-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('transform', needleRotationValue);
+    }
+
+    _populateDetailedResults(formattedValues) {
+        const resultsTables = document.querySelectorAll('.results-table');
+        let i = 0;
+        resultsTables[0].innerHTML = '';
+        for (; i < Math.ceil(formattedValues.length / 2); i++)
+            this._addDetailedResultsRow(resultsTables[0], i, formattedValues[i]);
+        resultsTables[1].innerHTML = '';
+        for (; i < formattedValues.length; i++)
+            this._addDetailedResultsRow(resultsTables[1], i, formattedValues[i]);
+    }
+
+    prepareUI() {
+        window.addEventListener('popstate', this._popStateHandler.bind(this), false);
+        window.addEventListener('resize', this._resizeScreeHandler.bind(this));
+        this._resizeScreeHandler();
+
+        document.getElementById("logo").onclick = this._logoClickHandler.bind(this);
+        document.getElementById("show-summary").onclick = (e) => this.showResultsSummary();
+        document.getElementById("show-details").onclick = (e) => this.showResultsDetails();
```

Comment:
Ditto

Reply:
Used a "cleaner" `.bind(this)` approach.